### ISSUE TITLE
Version 1.0.0 The example project crashes on RN 0.40.0

### ISSUE THREAD

**Issue by LiuCao** [TheWidlarzGroup/react-native-video#449]
I cloned the code submitted by 2017-01-11, running inside the example project, it would crash

**Comment by joekim** [comment]
@LiuCao I had a couple of crashers as well. 
In RCTVideo
1.  Sometimes source[@"type"] would be nil and would crash in setSrc:.
2. Sometimes self.onPlaybackRateChange would be nil and crash in observeValueForKeyPath:ofObject:change:context:

**Comment by smkhalsa** [comment]
I'm also experiencing this issue. Any recommended fixes?

**Comment by yushan023** [comment]
![image](https://cloud.githubusercontent.com/assets/20593532/21967880/a7291d70-dbc9-11e6-91b8-b1fef80e0d35.png)
How to download the version 1.0.0？   npm install， Can only be downloaded v0.9.0.

**Comment by LiuCao** [comment]
@yushan023 
```
npm i -S git://github.com/react-native-community/react-native-video.git

react-native link react-native-video
```

**Comment by LiuCao** [comment]
@joekim @smkhalsa 
Xcode shows the crash log is:
`
[__NSPlaceholderDictionary initWithObjects:forKeys:count:]: attempt to insert nil object from objects[1]'
`
The error means  trying to put nil in the dictionary (which is not allowed).

My fix method is ：
![fix ](https://raw.githubusercontent.com/LiuCao/HTStore/master/Issues/1.png)

**Comment by LiuCao** [comment]
@joekim @smkhalsa 
Xcode shows the crash log is:
`
[__NSPlaceholderDictionary initWithObjects:forKeys:count:]: attempt to insert nil object from objects[1]'
`
The error means  trying to put nil in the dictionary (which is not allowed).

My fix method is ：
![fix ](https://raw.githubusercontent.com/LiuCao/HTStore/master/Issues/1.png)

**Comment by joekim** [comment]
@smkhalsa  @LiuCao Yep, that's one of the two issues I encountered. The 2nd one is in observeValueForKeyPath:ofObject:change:context. I can push my fork with work arounds if that'll help. I'm not sure why source[@"type"] but I just put some protections in case it was.

I have one other change and that's to expose all the ivars and methods in the RCTVideo.h. The reason why is because I implemented a "fast" scrubber and it's easier to swizzle methods if everything is exposed via the header.

**Comment by abrahambotros** [comment]
@LiuCao, thanks for the suggestion. I put together your Objective C changes with an additional JS change from @smkhalsa above; see https://github.com/react-native-community/react-native-video/pull/455. What do you guys think of combining everything for this particular fix there in one place?

EDIT: As @joekim pointed out below, https://github.com/react-native-community/react-native-video/pull/455 only attempts to fix the `source[@type] = nil` NSDictionary error, and does not address the `self.onPlaybackRateChange` error mentioned above. As in my comment below, maybe https://github.com/react-native-community/react-native-video/pull/444 could help with this second error?

**Comment by joekim** [comment]
@abrahambotros How about the crasher where "self.onPlaybackRateChange" can be nil and crash in "observeValueForKeyPath:ofObject:change:context:"?

**Comment by abrahambotros** [comment]
@joekim, good point, sorry for dismissing that. I haven't been able to replicate that error on my setup yet, but it looks like maybe people are discussing the same error here: https://github.com/react-native-community/react-native-video/pull/444. I'll edit my comment above to note that I was only addressing the `source[@type] = nil` NSDictionary error. Thanks for catching that!

**Comment by hellogerard** [comment]
I think #444 and #455 are both critical and will hopefully go in the next release.

**Comment by crysfel** [comment]
I have the same problem.

Here's how I'm using the component:

```
<Video
        source={{ uri: 'https://someurl.com/files/file.mp3' }}
        rate={1.0}
        volume={1.0}
        paused={!playing}
        resizeMode="cover"
        progressUpdateInterval={1000}
        onLoadStart={onLoadStart}
        onLoad={this.setDuration}
        onProgress={this.setTime}
        onEnd={this.onEnd}
        playInBackground
        playWhenInactive
      />
```

The I got this error:
```
*** Terminating app due to uncaught exception 'NSInvalidArgumentException',
reason: '*** -[__NSPlaceholderDictionary initWithObjects:forKeys:count:]:
 attempt to insert nil object from objects[1]'
```

RN: 0.40
RBVideo: master

**Comment by mattapperson** [comment]
This issue should be fixed now in master and in the newly minted 1.0 release
